packages/sirius-web/backend/sirius-web-infrastructure/src/main/java/org/eclipse/sirius/web/infrastructure/elasticsearch/services/EditingContextIndexingService.java

```diff
@@ -0,0 +1,111 @@
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IIdentityService;
+import org.eclipse.sirius.components.emf.services.api.IEMFEditingContext;
+import org.eclipse.sirius.web.application.project.services.api.IProjectEditingContextService;
+import org.eclipse.sirius.web.infrastructure.elasticsearch.services.api.IEditingContextIndexingService;
+import org.eclipse.sirius.web.infrastructure.elasticsearch.services.api.IIndexDocumentCreationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch._helpers.bulk.BulkIngester;
+import co.elastic.clients.util.BinaryData;
+import co.elastic.clients.util.ContentType;
+
+/**
+ * Indexes the content of an editing context.
+ *
+ * @author gdaniel
+ */
+@Service
+public class EditingContextIndexingService implements IEditingContextIndexingService {
+
+    private final IProjectEditingContextService projectEditingContextService;
+
+    private final IIdentityService identityService;
+
+    private final IIndexDocumentCreationService indexDocumentCreationService;
+
+    private final Optional<ElasticsearchClient> optionalElasticSearchClient;
+
+    private final Logger logger = LoggerFactory.getLogger(EditingContextIndexingService.class);
+
+    public EditingContextIndexingService(IProjectEditingContextService projectEditingContextService, IIdentityService identityService, IIndexDocumentCreationService indexDocumentCreationService,
+            Optional<ElasticsearchClient> optionalElasticSearchClient) {
+        this.projectEditingContextService = Objects.requireNonNull(projectEditingContextService);
+        this.identityService = Objects.requireNonNull(identityService);
+        this.indexDocumentCreationService = Objects.requireNonNull(indexDocumentCreationService);
+        this.optionalElasticSearchClient = Objects.requireNonNull(optionalElasticSearchClient);
+    }
+
+    @Override
+    public void index(IEditingContext editingContext) {
+        long start = System.nanoTime();
+        if (this.optionalElasticSearchClient.isPresent()) {
+            if (editingContext instanceof IEMFEditingContext emfEditingContext) {
+                Optional<String> optionalProjectId = this.projectEditingContextService.getProjectId(editingContext.getId());
+                if (optionalProjectId.isPresent()) {
+                    this.clearIndex(optionalProjectId.get());
+                    BulkIngester<Void> bulkIngester = BulkIngester.of(bulkIngesterBuilder -> bulkIngesterBuilder
+                            .client(this.optionalElasticSearchClient.get())
```

Comment:
I kept the default configuration of the `BulkIngester`, meaning that bulk requests will be sent every 1000 operations or 5MB of data.
This means that even if the indexing takes some time, parts of the model will be indexed (and searchable) before the end of the operation.